src/v/model/record.h

```diff
@@ -417,11 +417,16 @@ struct record_batch_header {
 };
 
 using tx_seq = named_type<int64_t, struct tm_tx_seq>;
-
+using producer_id = named_type<int64_t, struct producer_identity_id>;
+using producer_epoch = named_type<int16_t, struct producer_identity_epoch>;
 struct producer_identity {
     int64_t id{-1};
     int16_t epoch{0};
```

Comment:
why not strongly type this `producer_epoch` ?

Reply:
and the id. as well.

Reply:
I'm in the process of doing it, this PR moves producer_id/producer_epoch from rm_stm, next PR will retype id&epoch. I don't want to do it right now because it will force me to substantively change future code making the scope too large.